protocol/messenger_communities.go

```diff
@@ -1921,3 +1914,389 @@ func (m *Messenger) RequestExtractDiscordChannelsAndCategories(filesToImport []s
 			errors)
 	}()
 }
+
+func (m *Messenger) RequestImportDiscordCommunity(request *requests.ImportDiscordCommunity) {
+	go func() {
+
+		importProgress := &discord.ImportProgress{}
+		importProgress.Init([]discord.ImportTask{
+			discord.CommunityCreationTask,
+			discord.ChannelsCreationTask,
+			discord.ImportMessagesTask,
+			discord.DownloadAssetsTask,
+			discord.InitCommunityTask,
+		})
+		importProgress.CommunityName = request.Name
+
+		exportData, errs := m.ExtractDiscordDataFromImportFiles(request.FilesToImport)
+		if len(errs) > 0 {
+			for _, err := range errs {
+				importProgress.AddTaskError(discord.CommunityCreationTask, err)
+			}
+			m.publishImportProgress(importProgress)
+			return
+		}
+		totalChannelsCount := len(exportData.ExportedData)
+		totalMessageCount := exportData.MessageCount
+
+		if totalChannelsCount == 0 || totalMessageCount == 0 {
+			importError := discord.Error(discord.ErrNoChannelData.Error())
+			if totalMessageCount == 0 {
+				importError.Message = discord.ErrNoMessageData.Error()
+			}
+			importProgress.AddTaskError(discord.CommunityCreationTask, importError)
+			importProgress.StopTask(discord.CommunityCreationTask)
+			m.publishImportProgress(importProgress)
+			return
+		}
+
+		importProgress.UpdateTaskProgress(discord.CommunityCreationTask, 0.5)
```

Comment:
Setting the progress to `0.5` is somewhat arbitrary here. I split up the `CommunityCreationTask` into a few parts and extracting the data accounts for `50%` of the work.

Happy to change this to something more reasonable.